internal/resource/config/oauth/accesstokenmapping/oauth_access_token_mappings_resource.go

```diff
@@ -0,0 +1,316 @@
+	state.Context, objDiags = types.ObjectValueFrom(ctx, accessTokenMappingContext, r.Context)
+	diags.Append(objDiags...)
+	state.AccessTokenManagerRef, objDiags = resourcelink.ToState(ctx, &r.AccessTokenManagerRef)
+	diags.Append(objDiags...)
+	state.AttributeSources, objDiags = attributesources.ToState(ctx, r.AttributeSources)
+	diags.Append(objDiags...)
+	state.AttributeContractFulfillment, objDiags = attributecontractfulfillment.ToState(ctx, r.AttributeContractFulfillment)
+	diags.Append(objDiags...)
+	state.IssuanceCriteria, objDiags = issuancecriteria.ToState(ctx, r.IssuanceCriteria)
+	diags.Append(objDiags...)
+
+	// make sure all object type building appends diags
+	return diags
+}
+
+func addOptionalOauthAccessTokenMappingsFields(ctx context.Context, addRequest *client.AccessTokenMapping, plan oauthAccessTokenMappingsResourceModel) error {
+	var err error
+	if internaltypes.IsDefined(plan.AttributeSources) {
+		addRequest.AttributeSources = []client.AttributeSourceAggregation{}
+		addRequest.AttributeSources, err = attributesources.ClientStruct(plan.AttributeSources)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.IssuanceCriteria) {
+		addRequest.IssuanceCriteria, err = issuancecriteria.ClientStruct(plan.IssuanceCriteria)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *oauthAccessTokenMappingsResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var model oauthAccessTokenMappingsResourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+	if internaltypes.IsDefined(model.Context) {
+		modelContextType := model.Context.Attributes()["type"].(types.String).ValueString()
+		modelContextContextRef := model.Context.Attributes()["context_ref"].(types.Object)
+		if (modelContextType == "DEFAULT" || modelContextType == "CLIENT_CREDENTIALS") && internaltypes.IsDefined(modelContextContextRef) {
+			resp.Diagnostics.AddError("Invalid attribute combination",
+				"context_ref is not required for the Access Token Mapping Context type: "+modelContextType)
+		}
+	}
+}
+
+func (r *oauthAccessTokenMappingsResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan oauthAccessTokenMappingsResourceModel
+	var err error
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	accessTokenMappingContext := &client.AccessTokenMappingContext{}
+	err = json.Unmarshal([]byte(internaljson.FromValue(plan.Context, true)), accessTokenMappingContext)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to create context object for request object:", err.Error())
+		return
+	}
+
+	accessTokenManagerRef, err := resourcelink.ClientStruct(plan.AccessTokenManagerRef)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to create AccessTokenManagerRef for request object:", err.Error())
```

Comment:
Should there be a return here and in the below check? Or maybe a final HasError check after all these but before the API call?